python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -376,6 +383,45 @@ def processResults(self, science, matchedTemplate, difference, sources, table,
 
         return measurementResults
 
+    def removeBadSources(self, diaSources):
+        """Remove bad diaSources from the catalog.
+
+        Parameters
+        ----------
+        diaSources : `lsst.afw.table.SourceCatalog`
+            The catalog of detected sources.
+
+        Returns
+        -------
+        diaSources : `lsst.afw.table.SourceCatalog`
+            The updated catalog of detected sources, with any source that has a
+            flag in ``config.badSourceFlags`` set removed.
+        """
+        nBadTotal = 0
+        selector = np.ones(len(diaSources), dtype=bool)
+        for flag in self.config.badSourceFlags:
+            try:
+                flags = diaSources[flag]
+            except KeyError as e:
+                self.log.warning("Could not apply source flag: %s", e)
+                continue
+            nBad = np.count_nonzero(flags)
+            if nBad > 0:
+                self.log.info("Found and removed %d unphysical sources with flag %s.", nBad, flag)
+                selector &= ~flags
+                nBadTotal += nBad
+        self.metadata.add("nRemovedBadFlaggedSources", nBadTotal)
+        # Use slot_Centroid_x/y here instead of getX() method, since the former
+        #  works on non-contiguous source tables and the latter does not.
+        centroidFlag = np.isfinite(diaSources["slot_Centroid_x"]) & np.isfinite(diaSources["slot_Centroid_y"])
+        nBad = np.count_nonzero(~centroidFlag)
+        if nBad > 0:
+            self.log.info("Found and removed %d unphysical sources with non-finite centroid.", nBad)
+            self.metadata.add("nRemovedBadCentroidSources", nBadTotal)
+            nBadTotal += nBad
+            selector &= centroidFlag
```

Comment:
You should be able to replace this entire thing with adding `slot_Centroid_flag` to the bad flags list. See DM-7102: that flag should be catching all cases where the centroid positions are non-finite.

Reply:
I want to explicitly catch `diaSources["slot_Centroid_x"]` (and `y`) here since that breaks downstream code. As @parejkoj and I discussed, this check should be unnecessary after DM-42313 merges since these nan centroids would then be picked up by the flag `base_PixelFlags_flag_offimage`.